lib/UndoTruncateToOddIntegerPass.cpp

```diff
@@ -160,31 +206,28 @@ bool UndoTruncatedSwitchConditionPass::runOnModule(Module &M) {
   zombies_.reset();
   i32_ = Type::getInt32Ty(M.getContext());
 
-  for (auto SI : WorkList) {
-    auto Cond = SI->getCondition();
-
-    auto *widened = ZeroExtend(Cond);
-    SI->setCondition(widened);
-    Changed = true;
+  while (!WorkList.empty()) {
+    auto inst = WorkList.back();
+    WorkList.pop_back();
 
-    for (auto Cases : SI->cases()) {
-      // The original value of the case.
-      auto V = Cases.getCaseValue()->getZExtValue();
+    auto extended = ZeroExtend(inst);
+    if (extended && extended != inst) {
+      Changed = true;
 
-      // A new value for the case with the correct type.
-      auto CI = dyn_cast<ConstantInt>(ConstantInt::get(i32_, V));
-
-      // And we replace the old value.
-      Cases.setValue(CI);
+      for (auto user : inst->users()) {
+        if (auto user_inst = dyn_cast<Instruction>(user)) {
+          WorkList.push_back(user_inst);
+        }
+      }
     }
   }
 
-  // Remove the zombies if we can.  We expect to.  They are ordered from
-  // combinations down to their supporting values.
-  for (auto *zombie : zombies_) {
-    if (!zombie->hasNUsesOrMore(1)) {
+  // Remove the zombies if we can.  We expect to. We've ordered zombies in
+  // reverse.
+  for (int i = zombies_.size(); i >= 1; --i) {
+    auto zombie = zombies_[i];
+    if (!zombie->hasNUsesOrMore(1))
```

Comment:
Arguably this should just be an assertion; if there's still uses then something has gone wrong.